src/components/organisms/AdminVenueView/components/Spaces/Spaces.tsx

```diff
@@ -146,17 +92,31 @@ export const Spaces: React.FC<SpacesProps> = ({ venue, onClickNext }) => {
     [venue.rooms]
   );
 
+  const onAdd = useCallback(
+    ({ data }) => {
+      const created =
+        venue.rooms?.[data?.roomIndex] ??
+        (data?.rooms)[data?.roomIndex] ??
+        data?.provided?.room;
+      if (!created) return;
+
+      setSelectedRoom(created);
+    },
+    [setSelectedRoom, venue.rooms]
+  );
+
```

Comment:
The idea behind this is to work out which is the newly added room from the most relevant source of the information and fall back to the least one. In this case that's the `venue.rooms` that would generally be in-sync through a connect hook.